### PR TITLE
feat(base/native): button abstractions

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react/features/base/react/components/native/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import styles from './styles';
+import { BUTTON_MODES, BUTTON_TYPES } from '../../constants';
+
+interface ButtonProps {
+    accessibilityLabel?: any;
+    color?: string;
+    disabled?: boolean;
+    icon?: any;
+    label?: any;
+    labelStyle?: any;
```
Comment (saghul) [inline react/features/base/react/components/native/Button.tsx:15]: Do we have any case where we'd want to use this prop?

File: react/features/base/react/components/native/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    mode?: any;
```
Comment (saghul) [inline react/features/base/react/components/native/Button.tsx:16]: Remove this one, as discussed, mode should be set based on the type.

File: react/features/base/react/components/native/IconButton.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    } else if (type === SECONDARY) {
+        color = BaseTheme.palette.icon02;
+        iconButtonContainerStyles = styles.iconButtonContainerSecondary;
+        rippleColor = BaseTheme.palette.action02;
+    } else {
+        color = iconColor;
+        rippleColor = tapColor;
+    }
+
+    if (mode === ICON) {
```
Comment (saghul) [inline react/features/base/react/components/native/IconButton.tsx:55]: What is this mode?

File: react/features/base/react/constants.js [hunk 4]
```diff
@@ -5,3 +5,21 @@
...
  */
 export const OVERLAY_Z_INDEX = 1000;
+
+/**
+ * The types of the buttons.
+ */
+export const BUTTON_TYPES = {
+    PRIMARY: 'primary',
+    SECONDARY: 'secondary',
+    DESTRUCTIVE: 'destructive'
```
Comment (saghul) [inline react/features/base/react/constants.js:15]: wasn't there a terciary one? Or re we net using it?

File: react/features/base/react/constants.js [hunk 5]
```diff
@@ -5,3 +5,21 @@
...
+};
+
+/**
+ * The modes of the buttons.
+ */
+export const BUTTON_MODES = {
```
Comment (saghul) [inline react/features/base/react/constants.js:21]: Drop this?

File: react/features/conference/components/native/carmode/Conference.tsx [hunk 6]
```diff
@@ -66,15 +81,6 @@ const CarmodeTab = (): JSX.Element => {
                 style = { styles.microphoneContainer }>
                 <MicrophoneButton />
             </View>
-            <View
```
Comment (saghul) [inline react/features/conference/components/native/carmode/Conference.tsx:69]: Why this change?

File: react/features/prejoin/components/Prejoin.native.tsx [hunk 7]
```diff
@@ -32,11 +32,11 @@ import { isDisplayNameRequired } from '../functions';
 import styles from './styles';
 
 
-interface Props {
+interface PrejoinProps {
     navigation: any;
```
Comment (saghul) [inline react/features/prejoin/components/Prejoin.native.tsx:36]: Can't we add the types for this?

File: react/features/reactions/components/native/RaiseHandButton.js [hunk 8]
```diff
@@ -170,8 +173,7 @@ function _mapStateToProps(state): Object {
     return {
         _enabled: enabled,
         _localParticipant,
-        _raisedHand: hasRaisedHand(_localParticipant),
-        _styles: ColorSchemeRegistry.get(state, 'Toolbox').raiseHandButton
```
Comment (saghul) [inline react/features/reactions/components/native/RaiseHandButton.js:174]: Can you drop that `raisedHandButton` style from the registry?

File: react/features/reactions/components/native/styles.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import BaseTheme from '../../../base/ui/components/BaseTheme.native';
+
+/**
+ * The styles of the native components of the feature {@code reactions}.
+ */
+export default {
+
+    raiseHandButton: {
+        marginVertical: BaseTheme.spacing[3],
+        width: '100%'
```
Comment (saghul) [inline react/features/reactions/components/native/styles.js:10]: Please limit this, think of the ipad.

File: react/features/reactions/components/native/styles.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import BaseTheme from '../../../base/ui/components/BaseTheme.native';
+
+/**
+ * The styles of the native components of the feature {@code reactions}.
+ */
+export default {
+
```
Comment (saghul) [inline react/features/reactions/components/native/styles.js:7]: style: stray newline

File: react/features/base/react/components/native/IconButton.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    onPress?: any;
+    size?: number;
+    src: any;
+    style?: any;
+    tapColor?: string;
+    type?: string;
+}
+
+
+const IconButton: React.FC<ButtonProps> = ({
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/react/components/native/IconButton.tsx:24]: Ah I see now. We only use it on that "..." button. I don't like that we are creating an abstraction just for that. Let's sync with Tibi, I was suggesting we merge the mute all and ... buttons into a single "Moderation" button.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/reactions/components/native/RaiseHandButton.js (hunk 8 above)
Comment (Calinteodor) [reply react/features/reactions/components/native/RaiseHandButton.js:174]: I removed it. It already appears in the PR.

File: react/features/conference/components/native/carmode/Conference.tsx (hunk 6 above)
Comment (Calinteodor) [reply react/features/conference/components/native/carmode/Conference.tsx:69]: Because he have the footerComponent  that we can use.

File: react/features/base/react/components/native/IconButton.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/base/react/components/native/IconButton.tsx:24]: Let's sync. But I find this approach much easier to follow and manage. 
I don't think it complicates or hinders stuff.

File: react/features/base/react/constants.js (hunk 5 above)
Comment (Calinteodor) [reply react/features/base/react/constants.js:21]: I vote to keep this because I might need it.

File: react/features/reactions/components/native/RaiseHandButton.js (hunk 8 above)
Comment (saghul) [reply react/features/reactions/components/native/RaiseHandButton.js:174]: 👍

File: react/features/conference/components/native/carmode/Conference.tsx (hunk 6 above)
Comment (saghul) [reply react/features/conference/components/native/carmode/Conference.tsx:69]: It doesn't, you had to disable the linter!

File: react/features/base/react/components/native/IconButton.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/react/components/native/IconButton.tsx:24]: My point is we have a single usage of this. If that usage goes away, we can skip it.

File: react/features/base/react/constants.js (hunk 4 above)
Comment (Calinteodor) [reply react/features/base/react/constants.js:15]: Added it.

File: react/features/conference/components/native/carmode/Conference.tsx (hunk 6 above)
Comment (Calinteodor) [reply react/features/conference/components/native/carmode/Conference.tsx:69]: No longer. Please check the latest commits.

File: react/features/prejoin/types.ts [hunk 12]
```diff
@@ -0,0 +1,3 @@
+interface PrejoinProps {
+    navigation: any;
```
Comment (saghul) [inline react/features/prejoin/types.ts:2]: Can we add the type for this?

**Review by saghul — COMMENTED** [review]
A couple of minor comments, LGTM otherwise! Really love how good (and consistent) they look now!

File: react/features/conference/components/native/carmode/Conference.tsx [hunk 13]
```diff
@@ -41,8 +41,22 @@ const CarmodeTab = (): JSX.Element => {
         };
     }, []);
 
+    const footerComponent = () => (
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/conference/components/native/carmode/Conference.tsx:44]: Got it. Will create a separate PR.

File: react/features/prejoin/types.ts (hunk 12 above)
Comment (Calinteodor) [reply react/features/prejoin/types.ts:2]: Will check it.

**Review by saghul — DISMISSED** [review]
